<launch>
    <arg name="resolution" />
    <arg name="image_topic" default="/bebop/image_raw" />
    <arg name="center_of_scan_topic" default="/pose" />
    <arg name="image_topic_fps" default="30" />
    <arg name="ns" default="" />
    <node pkg="tf" type="static_transform_publisher" name="laser_to_base_tf" args="0 0 0 0 0 0 $(arg ns)/base_link $(arg ns)/contours_scan_link 50" />
    <node pkg="contour_based_localization" type="contours_laserscan_generator.py" name="contours_laserscan_generator" output="screen" >
        <remap from="/camera/image_raw" to="$(arg image_topic)" />
        <remap from="/center_of_scan" to="$(arg center_of_scan_topic)" />
        <param name="laser_scale_factor" value="$(arg resolution)" />
        <param name="image_fps" value="$(arg image_topic_fps)" />
        <param name="scan_frame_id" value="$(arg ns)/contours_scan_link" />
        <param name="laser_radius_pixels" value="230" />
        <!--<param name="laser_samples_number" value="500" />-->
    </node>
</launch>